src/sage/schemes/elliptic_curves/ell_point.py

```diff
@@ -2746,47 +2811,60 @@ def order(self):
             sage: P.additive_order()
             2
         """
-        try:
-            return self._order
-        except AttributeError:
-            pass
+        return super().order(algorithm)
 
-        if self.is_zero():
-            self._order = Integer(1)
-            return self._order
+    def _compute_order(self, algorithm):
+        """
+        TESTS::
 
+            sage: E = EllipticCurve([0,0,1,-1,0])
+            sage: P = E([0,0]); P
+            (0 : 0 : 1)
+            sage: P._compute_order('pari')
+            +Infinity
+            sage: P._compute_order('unknown')
+            Traceback (most recent call last):
+            ...
+            NotImplementedError: unknown algorithm 'unknown'
+        """
         E = self.curve()
 
-        # First try PARI
-        try:
+        if algorithm == 'pari':
             n = E.pari_curve().ellorder(self)
             if n:
-                n = Integer(n)
+                return Integer(n)
             else:
-                n = oo
-            self._order = n
-            return n
-        except PariError:
-            pass
+                return oo
 
-        # Get the torsion order if known, else a bound on (multiple
-        # of) the order.  We do not compute the torsion if it is not
-        # already known, since computing the bound is faster (and is
-        # also cached).
+        if algorithm == 'generic':
+            # Get the torsion order if known, else a bound on (multiple
+            # of) the order.  We do not compute the torsion if it is not
+            # already known, since computing the bound is faster (and is
+            # also cached).
+            try:
+                N = E._torsion_order
+            except AttributeError:
+                N = E._torsion_bound()
 
-        try:
-            N = E._torsion_order
-        except AttributeError:
-            N = E._torsion_bound()
+            # Now self is a torsion point iff it is killed by N:
+            if not (N*self).is_zero():
+                return oo
 
-        # Now self is a torsion point iff it is killed by N:
-        if not (N*self).is_zero():
-            self._order = oo
-            return self._order
+            # Finally we find the exact order using the generic code:
+            return generic.order_from_multiple(self, N, operation='+')
 
-        # Finally we find the exact order using the generic code:
-        self._order = generic.order_from_multiple(self, N, operation='+')
-        return self._order
+        if algorithm in ('generic_small', 'hybrid'):
+            return super()._compute_order(algorithm)
+
+        if algorithm is not None:
+            raise NotImplementedError(f"unknown algorithm {algorithm!r}")
+
+        # First try PARI
+        try:
+            return self._compute_order('pari')
+        except PariError:
+            pass
+        return self._compute_order('generic')
```

Comment:
I have no idea how can Pari error be raised. Arguably if this line is hit it would be a pari bug.

Reply:
yes:

```
F = GF(2^70)
E = EllipticCurve([F.random_element() for __ in range(5)])
E.0._compute_order(algorithm='pari')

F = GF(next_prime(2^128))
E = EllipticCurve([F.random_element() for __ in range(5)])
E.0._compute_order(algorithm='pari')

R.<x> = QQ[]
F.<a> = NumberField(x^2+next_prime(2^128))
E = EllipticCurve([F.random_element() for __ in range(5)])
E.0._compute_order(algorithm='pari')  # takes forever but it probably won't error out
```